### ISSUE TITLE
Documentation of Coding Standards

### ISSUE THREAD

**Issue by ragingpastry** [AFCYBER-DREAM/bashellite#46]
It would be nice to have a CONTRIBUTING.md document to define the process for contributing new changes to this project. 

Some things that we could document include:
  - bash coding standards. Whether this is a link to Google's Bash standards or something custom..
  - Git commit / branching standards
  - Some examples for setting up a development environment
  - Documentation on how the whole system is put together, from an architecture point of view.

**Comment by pcseanmckay** [comment]
Take a look at this article.  It is essentially what we are doing...
https://barro.github.io/2016/02/a-succesful-git-branching-model-considered-harmful/

**Comment by ragingpastry** [comment]
Yup that is the other article that I have been looking at as well. If that is essentially what you guys have been doing then we can get something written up in a CONTRIBUTING.md document to clearly define what a new change should look like.
